### PR TITLE
Enable `link-to-github-io` for `.github.com` repos

### LINKED ISSUE #6228: Link to github io only works on `.github.io` repos, but older ones are `.github.com`
### Description

GitHub Pages switched the name of the repo from `username.github.com` to `username.github.io` about 8 years ago. As a result [many .com repos are still active](https://github.com/search?q=github.com&type=repositories) even though they actually result it `.github.io` pages. For example https://github.com/resume/resume.github.com leads to https://resume.github.io/

The feature that puts a hyperlink to the associated website only works with the new repo-to-page name and disregards the `.github.com` repos.

### How to replicate the issue + URL

Open any old GitHub Pages repo like: https://github.com/resume/resume.github.com and observe that there is no link

### Extension version

any

### Browser(s) used

any

### PR DESCRIPTION
**Pull request by ambujsahu81**
1. closes [#6228](https://github.com/refined-github/refined-github/issues/6228)
2. Affects old GitHub Pages repo like: [resume/resume.github.com](https://github.com/resume/resume.github.com)

## Test URLs
[resume/resume.github.com](https://github.com/resume/resume.github.com)

## Screenshot
- Repo Header 
 
repoheader

- Repo List

repoListView

### CODE REVIEW

File: source/features/link-to-github-io.tsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ import observe from '../helpers/selector-observer';
 function getLinkToGitHubIo(repoTitle: HTMLElement, className?: string): JSX.Element {
 	return (
 		<a
-			href={`https://${repoTitle.textContent!.trim()}`}
+			href={`https://${repoTitle.textContent!.trim().replace('.com', '.io')}`}
```
Comment (FloEdelmann) [inline source/features/link-to-github-io.tsx:12]: Maybe to be safe:

```suggestion
			href={`https://${repoTitle.textContent!.trim().replace('.github.com', '.github.io')}`}
```

File: source/features/link-to-github-io.tsx [hunk 2]
```diff
@@ -30,15 +30,16 @@ function initRepo(signal: AbortSignal): void {
 }
 
 function initRepoList(signal: AbortSignal): void {
-	observe('a[href$=".github.io"][itemprop="name codeRepository"]', addRepoListLink, {signal});
+	// Also consider any old GitHub Pages repo like: resume/resume.github.com , issue: https://github.com/refined-github/refined-github/issues/6228
+	observe(':is(a[href$=".github.io"], a[href$=".github.com"])[itemprop="name codeRepository"]', addRepoListLink, {signal});
```
Comment (FloEdelmann) [inline source/features/link-to-github-io.tsx:34]:
```suggestion
	observe('a:is([href$=".github.io"], [href$=".github.com"])[itemprop="name codeRepository"]', addRepoListLink, {signal});
```

File: source/features/link-to-github-io.tsx [hunk 3]
```diff
@@ -30,15 +30,16 @@ function initRepo(signal: AbortSignal): void {
...
 }
 
 void features.add(import.meta.url, {
 	include: [
 		pageDetect.hasRepoHeader,
 	],
 	asLongAs: [
-		() => Boolean(getRepo()?.name.endsWith('.github.io')),
+		() => Boolean(getRepo()?.name.endsWith('.github.io')) || Boolean(getRepo()?.name.endsWith('.github.com')),
```
Comment (FloEdelmann) [inline source/features/link-to-github-io.tsx:42]: Can we avoid the duplicate function call here please?

File: source/features/link-to-github-io.tsx (hunk 2 above)
Comment (ambujsahu81) [reply source/features/link-to-github-io.tsx]: I have updated this too. Thanks for reviewing and suggestion 🙂

File: source/features/link-to-github-io.tsx (hunk 3 above)
Comment (ambujsahu81) [reply source/features/link-to-github-io.tsx]: @FloEdelmann  i have updated it like this `Boolean(getRepo()?.name.endsWith('.github.io') || getRepo()?.name.endsWith('.github.com'))`. But not sure if this is what you were expecting.

File: source/features/link-to-github-io.tsx (hunk 2 above)
Comment (fregante) [reply source/features/link-to-github-io.tsx]: You can "Commit suggestion" in the future as well, if they don't need further changes

File: source/features/link-to-github-io.tsx (hunk 3 above)
Comment (FloEdelmann) [reply source/features/link-to-github-io.tsx]: Sorry for being unclear. I meant we should avoid calling `getRepo()` twice here.

File: source/features/link-to-github-io.tsx [hunk 4]
```diff
@@ -30,15 +30,16 @@ function initRepo(signal: AbortSignal): void {
...
+	observe('a:is([href$=".github.io"], [href$=".github.com"])[itemprop="name codeRepository"]', addRepoListLink, {signal});
 }
 
 void features.add(import.meta.url, {
 	include: [
 		pageDetect.hasRepoHeader,
 	],
 	asLongAs: [
-		() => Boolean(getRepo()?.name.endsWith('.github.io')),
+		() => Boolean(getRepo()?.name.endsWith('.github.io') || getRepo()?.name.endsWith('.github.com')),
```
Comment (fregante) [inline source/features/link-to-github-io.tsx:42]: Let's use a regex instead
```suggestion
		() => /\.github\.(io|com).test(getRepo()?.name),
```

File: source/features/link-to-github-io.tsx [hunk 5]
```diff
@@ -30,15 +30,16 @@ function initRepo(signal: AbortSignal): void {
 }
 
 function initRepoList(signal: AbortSignal): void {
-	observe('a[href$=".github.io"][itemprop="name codeRepository"]', addRepoListLink, {signal});
+	// Also consider any old GitHub Pages repo like: resume/resume.github.com , issue: https://github.com/refined-github/refined-github/issues/6228
+	observe('a:is([href$=".github.io"], [href$=".github.com"])[itemprop="name codeRepository"]', addRepoListLink, {signal});
```
Comment (fregante) [inline source/features/link-to-github-io.tsx:34]:
```suggestion
	observe(`
		a[itemprop="name codeRepository"]:is(
			[href$=".github.io"],
			[href$=".github.com"]
		)
	`, addRepoListLink, {signal});
```

File: source/features/link-to-github-io.tsx [hunk 6]
```diff
@@ -30,15 +30,16 @@ function initRepo(signal: AbortSignal): void {
 }
 
 function initRepoList(signal: AbortSignal): void {
-	observe('a[href$=".github.io"][itemprop="name codeRepository"]', addRepoListLink, {signal});
+	// Also consider any old GitHub Pages repo like: resume/resume.github.com , issue: https://github.com/refined-github/refined-github/issues/6228
```
Comment (fregante) [inline source/features/link-to-github-io.tsx:33]:
```suggestion
	// Earlier GitHub Pages were hosted on github.com #6228
```

File: source/features/link-to-github-io.tsx [hunk 7]
```diff
@@ -9,7 +9,7 @@ import observe from '../helpers/selector-observer';
 function getLinkToGitHubIo(repoTitle: HTMLElement, className?: string): JSX.Element {
 	return (
 		<a
-			href={`https://${repoTitle.textContent!.trim()}`}
+			href={`https://${repoTitle.textContent!.trim().replace('.github.com', '.github.io')}`}
```
Comment (fregante) [inline source/features/link-to-github-io.tsx:12]:
```suggestion
			href={`https://${repoTitle.textContent!.trim().replace(/com$/, 'io')}`}
```

**Review by fregante — APPROVED** [review]

File: source/features/link-to-github-io.tsx [hunk 8]
```diff
@@ -30,15 +30,21 @@ function initRepo(signal: AbortSignal): void {
...
 
 function initRepoList(signal: AbortSignal): void {
-	observe('a[href$=".github.io"][itemprop="name codeRepository"]', addRepoListLink, {signal});
+	// Earlier GitHub Pages were hosted on github.com #6228
+	observe(`
+		a[itemprop="name codeRepository"]:is(
+			[href$=".github.io"],
+			[href$=".github.com"]
+		)
+	`, addRepoListLink, {signal});
```
Comment (fregante) [inline source/features/link-to-github-io.tsx:39]: Interestingly this broke our selector parser https://github.com/g-plane/typed-query-selector/issues/26

Let's just use an array

```suggestion
	observe([
		// Earlier GitHub Pages were hosted on github.com #6228
		'a[itemprop="name codeRepository"][href$=".github.com"]',
		'a[itemprop="name codeRepository"][href$=".github.io"]',
	], addRepoListLink, {signal});
```

File: source/features/link-to-github-io.tsx (hunk 3 above)
Comment (ambujsahu81) [reply source/features/link-to-github-io.tsx]: Thank you  @fregante  for making the changes, I wasn't able to solve this one. @FloEdelmann  again thanks for suggesting this 🙂

File: source/features/link-to-github-io.tsx [hunk 9]
```diff
@@ -30,15 +30,21 @@ function initRepo(signal: AbortSignal): void {
...
 }
 
 void features.add(import.meta.url, {
 	include: [
 		pageDetect.hasRepoHeader,
 	],
 	asLongAs: [
-		() => Boolean(getRepo()?.name.endsWith('.github.io')),
+		() => /\.github\.(io|com)$/.test(getRepo()?.name),
```
Comment (fregante) [inline source/features/link-to-github-io.tsx:47]:
```suggestion
		() => /\.github\.(io|com)$/.test(getRepo()?.name!),
```

File: source/features/link-to-github-io.tsx [hunk 10]
```diff
@@ -30,15 +30,19 @@ function initRepo(signal: AbortSignal): void {
...
+	], addRepoListLink, {signal});
 }
 
 void features.add(import.meta.url, {
 	include: [
 		pageDetect.hasRepoHeader,
 	],
 	asLongAs: [
-		() => Boolean(getRepo()?.name.endsWith('.github.io')),
+		() => /\.github\.(io|com)$/.test(getRepo()?.name!),
```
Comment (fregante) [inline source/features/link-to-github-io.tsx:45]:
```suggestion
		() => /\.github\.(io|com)$/.test(getRepo()?.name ?? 'shush ts'),
```

Comment (fregante) [reply source/features/link-to-github-io.tsx]: This is better than calling getRepo twice. Please use this solution

Comment (ambujsahu81) [reply source/features/link-to-github-io.tsx]: I have applied the suggested changes. 🙂

**Comment by fregante** [comment]
Thank you @ambujsahu81! I think this is ready for merge. I'll just keep it open a few days since we just released a new version and I'm not sure when the next one will be.

**Comment by ambujsahu81** [comment]
> Thank you @ambujsahu81! I think this is ready for merge. I'll just keep it open a few days since we just released a new version and I'm not sure when the next one will be.

@fregante that is fine , thank you for reviewing and suggestions .🙂

**Comment by fregante** [comment]
Tested, it works!  Thank you!

For future PRs, make sure you use a different branch every time to avoid checkout issues 😅

**Pull request merged**